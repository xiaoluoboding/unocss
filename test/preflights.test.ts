import { createGenerator } from '@unocss/core'
import { describe, expect, test } from 'vitest'

describe('preflights', () => {
  test('basic', async() => {
    const uno = createGenerator({
      preflights: [
        {
          getCSS() {
            return '.hello { text: red }'
          },
          layer: 'preflight',
        },
        {
          getCSS() { return undefined },
          layer: 'void',
        },
        {
          getCSS() { return '.default-preflight {}' },
        },
      ],
      presets: [],
    })
    const { css } = await uno.generate('')
    expect(css).toMatchSnapshot()
  })
})
