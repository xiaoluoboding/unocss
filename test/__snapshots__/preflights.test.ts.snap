// Vitest Snapshot v1

exports[`preflights > basic 1`] = `
"/* layer: default */
.default-preflight {}

/* layer: preflight */
.hello { text: red }
"
`;
